target_snowflake/__init__.py

```diff
@@ -170,6 +171,31 @@ def persist_lines(config, lines, table_cache=None, file_format_type: FileFormatT
             else:
                 records_to_load[stream][primary_key_string] = o['record']
 
+            if config.get('archive_load_files.enabled'):
+                # keep track of min and max
+                archive_load_files_primary_column =\
+                    config.get('archive_load_files.primary_column') or primary_key_string
+                archive_primary_column_value =\
+                    records_to_load[stream][primary_key_string][archive_load_files_primary_column]
+                values = archive_load_files_data[stream] or {
+                    's3_key': 'tbd',  # TODO: Turn archive_load_files.naming_convention into actual key
+                    'database': 'tbd',  # TODO: Where to get this
+                    'schema': 'tbd',  # TODO: Where to get this
+                    'table': stream,
```

Comment:
these values, especiall`s3_key` should be defined at an another place level when loading into s3.

The stream is a concatenated string of db-schema and db-table and that's probably enough  to pass furhter to other functions. The `database` is  not acceptable in general in target-snowflake. One ppw stream can read only from one database and DB-schema-table hierarchy is not acceptable to every db type.  (pg vs mariadb)

Reply:
@koszti One issue with `s3_key` is that it requires access to the config. From here, currently, `config` is not passed further than `flush_streams`. Would you prefer to have the entire config propagated further from there, or add the `archive_load_files` config as init param to the `S3UploadClient`, or something else?